retriever/lib/engine.py

```diff
@@ -489,25 +491,72 @@ def download_file(self, url, filename):
                     stream=True,
                     headers={
                         'user-agent':
-                            'Weecology/Data-Retriever \
-                                            Package Manager: http://www.data-retriever.org/'
+                            'Weecology/Data-Retriever Package Manager: http://www.data-retriever.org/'
                     },
-                    hooks={'response': reporthook(progbar, path)},
-                )
+                    hooks={'response': reporthook(progbar, path)})
 
                 if response.status_code == 404:
                     print("The data source or server may be redirected or not found")
+                    return False
 
             except InvalidSchema:
                 try:
                     urlretrieve(url, path, reporthook=reporthook(progbar))
                 except HTTPError as e:
                     print("HTTPError :", e)
-                    return None
+                    return False
+
+            return True
 
-            self.use_cache = True
-            progbar.close()
-        return True
+        except:
+            print("The download url {} is incorrect!!".format(url))
+            return False
+
+    def download_file(self, url, filename):
```

Comment:
Let's move `def download_file` above `def download_from_socrata`. We try to cluster functionality together but also sort where possible